ui-app/client/src/commonComponents/MarkDownNavigatorLinks.tsx

```diff
@@ -0,0 +1,308 @@
+			let title = '';
+			if (
+				spaceIndex !== -1 &&
+				linkSubStr[spaceIndex + 1] === '"' &&
+				linkSubStr[linkSubStr.length - 1] === '"'
+			) {
+				title = linkSubStr.slice(spaceIndex + 2, linkSubStr.length - 1);
+			}
+			const actualLink = linkSubStr.slice(0, spaceIndex !== 1 ? spaceIndex : undefined);
+			tokens.push(
+				React.createElement('img', {
+					key: linkSubStr,
+					title: title ?? '',
+					src: actualLink,
+					loading: 'lazy',
+					alt: altText,
+				}),
+			);
+			tokenCount += 2;
+			i = linkEndIndex;
+		} else if (text[i] === '[') {
+			// link line
+			const endIndex = text.indexOf(']', i + 1);
+			if (endIndex === -1 || text[endIndex + 1] !== '(') {
+				tokens[tokenCount] = (tokens[tokenCount] ?? '') + text[i];
+				continue;
+			}
+			const linkText = text.slice(i + 1, endIndex);
+			const linkEndIndex = text.indexOf(')', endIndex);
+			if (linkEndIndex === -1) {
+				tokens[tokenCount] = (tokens[tokenCount] ?? '') + text[i];
+				continue;
+			}
+			let linkSubStr: string | string[] = text.slice(endIndex + 2, linkEndIndex);
+			const spaceIndex = linkSubStr.indexOf(' ');
+			let title = '';
+			if (
+				spaceIndex !== -1 &&
+				linkSubStr[spaceIndex + 1] === '"' &&
+				linkSubStr[linkSubStr.length - 1] === '"'
+			) {
+				title = linkSubStr.slice(spaceIndex + 2, linkSubStr.length - 1);
+			}
+			const actualLink = linkSubStr.slice(0, spaceIndex !== 1 ? spaceIndex : undefined);
+			tokens.push(
+				React.createElement(
+					'a',
+					{ key: linkSubStr[0], title: title ?? '', href: actualLink },
+					parseText(linkText),
+				),
+			);
+			tokenCount += 2;
+			i = linkEndIndex;
+		} else if (text[i] === '\\') {
+			// escape escape
+			tokens[tokenCount] = (tokens[tokenCount] ?? '') + text[i + 1];
+			i = i + 1;
+			continue;
+		} else {
+			// normal text
+			tokens[tokenCount] = (tokens[tokenCount] ?? '') + text[i];
+		}
+	}
+
+	return tokens;
+};
+
+// used to create the heading tags
+const getHeaderComp = (line: string) => {
+	let i = 0;
+
+	// iterate till the last #
+	while (line[i] === '#') {
+		i++;
+	}
+
+	// fetching the custom id "custom id starts with {#custom_id}"
+	const idIndex = line.indexOf('{#');
+	let id = '';
+	line = line.slice(i);
+	// now create a heading tag according to the count of #
+	const HeadingTag = `h${i}` as keyof JSX.IntrinsicElements;
+
+	if (idIndex === -1) {
+		// create new id
+		id = line
+			.replace(/[^\w\s]/gi, '')
+			.replace(/\s+/g, '')
+			.toLowerCase();
+	} else if (idIndex != -1 && line.indexOf('}', idIndex) !== -1) {
+		id = line.slice(idIndex + 2, line.indexOf('}', idIndex)).toLocaleLowerCase();
+		line = line.slice(0, idIndex) + line.slice(line.indexOf('}', idIndex) + 1);
+	}
+
+	const newProps = {
+		href: `#${id}`,
+		key: id,
+		style: {
+			display: 'block',
+			cursor: 'pointer',
+		},
+	};
+	return React.createElement(
+		'a',
+		newProps,
+		parseText(line).filter(e => !!e),
+	);
+};
+
+export default function MarkDownNavigatorLinks({
```

Comment:
Why is this in the Common Components?? Please move to MarkdownNavigator. I don't see using this in anyother components.